src/recipe_dictionary.cpp

```diff
@@ -33,9 +33,57 @@ std::vector<const recipe *> recipe_subset::search( const std::string &txt ) cons
 {
     std::vector<const recipe *> res;
 
+    size_t split;
+    split = txt.find( ":" );
+    std::string key = "", value;
+    if( split != std::string::npos ) {
+        key = txt.substr( 0, split );
+        value = txt.substr( split + 1 );
+    } else {
+        value = txt;
+    }
+
     std::copy_if( recipes.begin(), recipes.end(), std::back_inserter( res ),
-    [ &txt ]( const recipe * r ) {
-        return lcmatch( item::nname( r->result ), txt );
+    [&]( const recipe * r ) {//sadly it seems easier to copy the scope here for the moment
+        if( key == _( "Q" ) ) { // result quality match
+            const auto &result = item( r->result );
+            return std::any_of( result.type->qualities.begin(), result.type->qualities.end(),
+            [value]( const std::pair<quality_id, int> &quality ) {
+                return lcmatch( quality.first->name, value );
+            } );
+        } else if( key == _( "t" ) ) { // tool match
+            const auto &toolsets =
+                r->requirements().get_tools();
+            return std::any_of( toolsets.begin(), toolsets.end(),
+            [value]( const std::vector<tool_comp> &tool_set ) {
+                return std::any_of( tool_set.begin(), tool_set.end(),
+                [value]( const tool_comp & t ) {
+                    return lcmatch( t.to_string(), value );
+                } );
+            } );
+        } else if( key == _( "s" ) ) { // skill match
+            return lcmatch( r->required_skills_string(), value );
+        } else if( key == _( "q" ) ) { // requirement quality
+            const auto &req_sets = r->requirements().get_qualities();
+            return std::any_of( req_sets.begin(), req_sets.end(),
+            [value]( const std::vector<quality_requirement> &req_set ) {
+                return std::any_of( req_set.begin(), req_set.end(),
+                [value]( const quality_requirement & qr ) {
+                    return lcmatch( qr.to_string(), value );
+                } );
+            } );
+        } else if( key == _( "c" ) ) { // component match
+            const auto &required_comp = r->requirements().get_components();
```

Comment:
Can the indexed map do partial matching? I think partial mapping is somewhat important here.

At least, I really don't think that there's any substring matching we can do here with the sorted map (or an unsorted one).

That's something I want because I prefer being able to search for "string" instead of "long string".

Do you think that its worthwhile?